crates/stages/src/stages/headers.rs

```diff
@@ -76,6 +76,13 @@ impl<DB: Database, D: HeaderDownloader, C: Consensus, H: HeadersClient, S: Statu
 
         // Lookup the head and tip of the sync range
         let (head, tip) = self.get_head_and_tip(tx, current_progress).await?;
+
+        // Nothing to sync
+        if head.hash() == tip {
```

Comment:
What was the behavior before?

Reply:
this was unreachable because of `forkchoice.head_block_hash != *head` check in `next_fork_choice_state`